stdlib/src/random/random.mojo

```diff
@@ -201,3 +201,28 @@ fn randn[
     for i in range(size):
         ptr[i] = randn_float64(mean, variance).cast[type]()
     return
+
+
+fn shuffle[T: CollectionElement](inout list_in: List[T]):
+    """Shuffles the elements of the list randomly.
+
+    Performs an in-place Fisher-Yates shuffle on the provided list.
+
+    Args:
+        list_in: The list to modify.
+
+    Parameters:
+        T: The type of element in the List.
+    """
+
+    var length = len(list_in)
+
+    for i in range(length - 1, 0, -1):
+        var j: Int = random_ui64(0, i).cast[DType.int64]().value
+
+        var i_ptr = list_in.data.offset(i)
+        var j_ptr = list_in.data.offset(j)
+
+        var tmp = i_ptr.take_pointee()
```

Comment:
Can we use `swap` here?

Reply:
Unfortunately not.

Reply:
What about `swap(i_ptr[], j_ptr[])`?

Reply:
Actually, `swap(list.unsafe_get(i), list.unsafe_get(j))` could be even better.

Reply:
Very nice!